### PR TITLE
Visit Analysers

### PR DESCRIPTION
**Pull request by rait-mediarithmics**
This PR adds visit analysers in library and allows users to select a visit analyser settings/sites.

### CODE REVIEW

**Comment by rait-mediarithmics** [comment]
@depends on https://github.com/MEDIARITHMICS/core-platform-services/pull/148

File: app/src/core/settings/sites/edit.one.html [hunk 1]
```diff
@@ -67,10 +67,48 @@ <h3 id="aliases">Web Domain Aliases
...
+
+      <div class="row-fluid">
+          <div class="form-inline">
+              <span class="h3 ">Visit Analyser</span>
+              <span class="badge site right-spaced" data-html="true" uib-popover-html="'The visit analyser user for the domain'" uib-popover-template="" popover-trigger="mouseenter" popover-placement="right">?</span>
+               <div ng-include="'src/core/visitanalysers/choose-visit-analyser.html'" class="form-group"></div>
+          </div>
+          <br>
+
+          <table class="mics-table js-auto-hide-actions">
```
Comment (dduponchel) [inline app/src/core/settings/sites/edit.one.html]: You should hide the table without visit analyser.

File: app/src/core/settings/sites/edit.one.html [hunk 2]
```diff
@@ -67,10 +67,48 @@ <h3 id="aliases">Web Domain Aliases
       <mcs-event-rules rules="rules"></mcs-event-rules>
 
 
+      <br>
```
Comment (dduponchel) [inline app/src/core/settings/sites/edit.one.html]: If you use a real `<h3>` below, you won't need these `<br>` to emulate a css margin.

File: app/src/core/settings/sites/edit.one.html [hunk 3]
```diff
@@ -67,10 +67,48 @@ <h3 id="aliases">Web Domain Aliases
...
+              <th class="actions">Actions</th>
+            </tr>
+            </thead>
+            <tbody>
+            <tr fetch-plugin-properties-from-version="visitAnalyser.visit_analyzer_plugin_id as visitAnalyserProperties">
+              <td>{{visitAnalyser.name}}</td>
+              <td>{{visitAnalyserProperties['name'].value}}</td>
+              <td>{{visitAnalyserProperties['provider'].value}}</td>
+              <td class="actions">
+                <a class="mics-btn mics-btn-action" href="#/{{organisationId}}/library/visitanalysers/{{visitAnalyser.id}}">Edit</a>
```
Comment (dduponchel) [inline app/src/core/settings/sites/edit.one.html]: An edit button here looks like a bad idea, the edition should happens in the library page.

File: app/src/core/settings/sites/EditOneController.js [hunk 4]
```diff
@@ -179,6 +186,21 @@ define(['./module', 'jquery'], function (module, $) {
           }, handleSiteError);
         }
       };
+
+
+      // ---------------- VISIT ANALYSER ----------------
+      $scope.$on("mics-visit-analyser:selected", function (event, params) {
+        if (params.visitAnalyser === null) {
+          $scope.visitAnalyser  = undefined;
+          $scope.visitAnalyserProperties = undefined;
```
Comment (dduponchel) [inline app/src/core/settings/sites/EditOneController.js]: `visitAnalyserProperties` only exists in a child scope, setting it here is confusing.

**Comment by thomas-tosoni** [comment]
Ok for me :+1:

**Pull request merged**